### PR TITLE
lib: gui: lvgl: write once to mono displays

### PR DESCRIPTION
**Pull request by ghost**
For some reason the `lvgl_flush_cb_mono()` function interprets the double buffering flag as an instruction to write the display contents to the screen twice. AFAICT, this is not at all how double buffering works in lvgl. The other display types do not have this extra display update.

I'm using an e-paper display so the extra update is very noticeable, and avoiding it essentially halves the update time.

### CODE REVIEW

File: lib/gui/lvgl/lvgl_display_mono.c [hunk 1]
```diff
@@ -24,10 +24,6 @@ void lvgl_flush_cb_mono(struct _disp_drv_t *disp_drv,
 	desc.pitch = w;
```
Comment (kalvdans) [inline lib/gui/lvgl/lvgl_display_mono.c:24]: Unused variable; remove.

File: lib/gui/lvgl/lvgl_display_mono.c [hunk 2]
```diff
@@ -24,10 +24,6 @@ void lvgl_flush_cb_mono(struct _disp_drv_t *disp_drv,
...
 	desc.height = h;
 	display_write(display_dev, area->x1, area->y1, &desc, (void *) color_p);
-	if (cap.screen_info & SCREEN_INFO_DOUBLE_BUFFER) {
```
Comment (kalvdans) [inline lib/gui/lvgl/lvgl_display_mono.c:27]: This was the last use of `cap` so the variable could be removed.

File: lib/gui/lvgl/lvgl_display_mono.c [hunk 3]
```diff
@@ -24,10 +24,6 @@ void lvgl_flush_cb_mono(struct _disp_drv_t *disp_drv,
 	desc.pitch = w;
 	desc.height = h;
 	display_write(display_dev, area->x1, area->y1, &desc, (void *) color_p);
```
Comment (vanwinkeljan) [inline lib/gui/lvgl/lvgl_display_mono.c:26]: Note that `SCREEN_INFO_DOUBLE_BUFFER` in `screen_info` indicates that the display controller has two internal frame buffers, as `LVGL` only updates partial buffer space both buffers need to be updated else they are no longer in sync and the display will only show a partial result on the screen.

**Review by vanwinkeljan — CHANGES_REQUESTED** [review]
Proposed change will break `lvgl` sample for reelboard

**Review by jfischer-no — CHANGES_REQUESTED** [review]
Yes this would break gdeh0213b1 support (LVGL only). I spend some time to fix that in the driver, but gave it up. The _older_  (SSD1673) controller behaves differently, ideally we would describe it in DT with which driver is the display controller compatible, but it is not yet possible, see also https://github.com/zephyrproject-rtos/zephyr/issues/20287. Until then NACK.

File: lib/gui/lvgl/lvgl_display_mono.c (hunk 3 above)
Comment (weinholtendian) [reply lib/gui/lvgl/lvgl_display_mono.c:26]: I'm pretty sure that the ssd16xx display I'm using does not have two internal frame buffers. Each `display_write` call performs a "master activation" on the display, which is what causes the display to get updated from internal RAM. But it points me in the right direction for finding a proper fix, thank you very much.

**Comment by weinholtendian** [comment]
Thank you very much for the review everyone! I'm closing this PR since, even though it fixes my particular display, it's obviously the wrong fix in general.

**Pull request closed without merging**